### ISSUE TITLE
Proxy server with publicKey auth

### ISSUE THREAD

**Issue by ptorrent** [mscdex/ssh2#1262]
Hello there,

Is there a way to forward the publicKey to other ssh server ?

Something like that ?

```	
			 client.on('authentication', (ctx) => {
				else if(ctx.method == 'publickey'){
					connect({
						username 		: ctx.username,
						publicKey 		: ctx.key.data
					}, function(err, session){
						if(err) return ctx.reject()
						client.Weble = session 
						ctx.accept()
					})
				}
```

Not an expert, maybe it's crazy x)

Thanks for your help !

**Comment by mscdex** [comment]
No, there is currently no built-in way to proxy authentication to a third party.

**Comment by nodegin** [comment]
@mscdex I have the same question.
Could you give me some idea on how can I achieve something like this please?

For example, when user connects to ssh2 server,
we initiate a client request to another server immediately and proxy/pipe the incoming user connection to the outgoing client connection with another remote server.

Is this even technically possible?
If so it would be really helpful if you could give me some hint on how can I get started with editing the source to achieve this.
Great thanks for this awesome lib!
